### PR TITLE
add a view button to gitbook toolbar

### LINKED ISSUE #805: Feature request: option for "view on GitHub" button on gitbook toolbar
It would be great to have a button similar to "edit" that allows the reader to *view* the file on GitHub. When I'm reading bookdown books, 95% of the time I just want to see the source code, not edit it. I still click on the edit button but it takes a few additional clicks to see the file without having to fork the repository. Also, some people may be intimidated to click "edit" just to see the source code, so they would benefit from a separate button as well.

I realize I can change the behavior of the edit button to do what I want by changing the `edit:` link in `_bookdown.yml`, but don't want to do that in order to maintain consistency of buttons across all bookdown books.

---

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.name/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('bookdown')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('rstudio/bookdown')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

### PR DESCRIPTION
**Pull request by jtr13**
This closes #805. I followed the model of pull request #639 Add option for a `history` button.

### CODE REVIEW

**Comment by jtr13** [comment]
Here is an example of a bookdown book with the view button: https://jtr13.github.io/d3book/

File: inst/resources/gitbook/js/plugin-bookdown.js [hunk 1]
```diff
@@ -28,6 +28,18 @@ gitbook.require(["gitbook", "lodash", "jQuery"], function(gitbook, _, $) {
       }
     });
 
+        // add the View button (file view on Github)
+    var view = config.view;
+    if (view && view.link) gitbook.toolbar.createButton({
+      icon: 'fa fa-eye',
+      label: view.text || 'View',
```
Comment (yihui) [inline inst/resources/gitbook/js/plugin-bookdown.js]: How about `View` -> `View Source`?

Comment (jtr13) [reply inst/resources/gitbook/js/plugin-bookdown.js]: Yes, good suggestion.

**Comment by jtr13** [comment]
Fyi, DESCRIPTION still says Version: 0.15.1 but I didn't want to mess with that.

**Pull request merged**